File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.java
class MyQueue {
    Stack<Integer> stk1 = new Stack<Integer>();
    Stack<Integer> stk2 = new Stack<Integer>();    
    public MyQueue() {
        
    }
    
    public void push(int x) {
        if(stk1.isEmpty()){
            stk1.push(x);
        }
        else{
            while(!stk1.isEmpty()){
                stk2.push(stk1.pop());
            }
            stk1.push(x);

            while(!stk2.isEmpty()){
                stk1.push(stk2.pop());
            }
        }
        return;
    }
    
    public int pop() {
        return stk1.pop();
    }
    
    public int peek() {
        return stk1.peek();
         
    }
    
    public boolean empty() {
        if(stk1.isEmpty()){
            return true;
        }

        return false;
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: 0232-implement-queue-using-stacks/0232-implement-queue-using-stacks.py

class MyQueue(object):

    def __init__(self):
        self.s1 = []
        self.s2 = []

    def push(self, x):
        """
        :type x: int
        :rtype: None
        """
        s = len(self.s1)
        if len(self.s1) == 0:
            self.s1.append(x)
        elif not len(self.s1) == 0 and len(self.s2) == 0:
            while not len(self.s1) == 0:
                temp  = self.s1.pop()
                self.s2.append(temp)
            self.s1.append(x)

            while not len(self.s2) == 0:
                temp = self.s2.pop()
                self.s1.append(temp)

    def pop(self):
        """
        :rtype: int
        """
        return self.s1.pop()

    def peek(self):
        """
        :rtype: int
        """
        return self.s1[-1]

    def empty(self):
        """
        :rtype: bool
        """
        if len(self.s1) == 0:
            return True
        else:
            return False


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()